app/javascript/packages/document-capture/services/upload.ts

```diff
@@ -95,13 +98,6 @@ const upload: UploadImplementation = async function (payload, { method = 'POST',
     throw new Error(response.statusText);
   }
 
-  if (response.url !== endpoint) {
```

Comment:
Background: This was added in #5923 to handle CSRF failures for uploads. The way it worked relied on the backend triggering a redirect (i.e. responding with a URL different from what we sent the request to). This becomes an issue for the updated `request` code since the URL is changed by the request handler, and it was causing this code to try to redirect to the localized upload API route.

Instead, if we handle JSON directly for the CSRF error handler (which we should do anyways), we can return a JSON redirect value directly, which is already handled below on line 104.